HackStudio+TextEditor: Add statusbar with selected text information

ry-sev opened the pull request:
This removes the cursor label located at the top right of editor
and will implement a statusbar which displays the line
and column position of the cursor. When the user selects (highlights)
text, the statusbar will also show the number of selected lines,
words, and characters.

![2021-04-15_17-49](https://user-images.githubusercontent.com/38510174/114944941-7d161380-9e16-11eb-9299-b3e83a912a05.png)

ADKaster commented on Userland/DevTools/HackStudio/HackStudioWidget.cpp:
```diff
@@ -1083,6 +1089,46 @@ void HackStudioWidget::initialize_menubar(GUI::Menubar& menubar)
+        int word_count = 0;
+        bool in_word = false;
+        String selected_text = current_editor().selected_text();
+        for (char c : selected_text) {
```
Would it be better to iterate over this with a UTF-8 view and count code points instead?

FalseHonesty replied:
CLion seems to consider all emojis a single character in their selection length info, which seems like reasonable behavior imo

sin-ack replied:
The question is whether you count individual codepoints or grapheme clusters.

FalseHonesty replied:
Right, fair point, even CLion doesn't properly handle emojis like 👨‍👩‍👦 it seems. I think just counting codepoints is reasonable behavior for now?

ry-sev replied:
I'll be honest, I never heard of code points or grapheme clusters

linusg reviewed (commented):
This looks nice! I think it would be very useful to put the selection information "calculation" into `GUI::TextEditor` though so it can also be used in the TextEditor application and other programs, and to ensure they are all consistent with how they count things - what do you think?

ry-sev commented:
@linusgSure, I'll see what I can do

awesomekling commented on Userland/DevTools/HackStudio/HackStudioWidget.cpp:
```diff
@@ -1083,6 +1089,46 @@ void HackStudioWidget::initialize_menubar(GUI::Menubar& menubar)
+        if (first_line == last_line || selection.start().column() == 0 || selection.start().column() != 0)
+            last_line += 1;
+
+        builder.appendff("        Selected: {} {}", last_line - first_line, (last_line - first_line) != 1 ? "Lines" : "Line");
```
Please use multiple statusbar segments instead of adding a bunch of whitespace to the first segment :)

ry-sev commented:
![2021-04-17_18-43](https://user-images.githubusercontent.com/38510174/115128586-d617b000-9fac-11eb-91dc-84421a7ce0eb.png)

Updated screenshot of the statusbar with segments. I ended up putting 3 segments because two segments like kinda awkward because of the space between the texts. But I can change this to two segments if needed.

I also added functions to TextEditor which calculates and returns the number of selected lines and words. This is so it can be used by other editors, e.g TextEditor.

linusg commented on Userland/DevTools/HackStudio/HackStudioWidget.cpp:
```diff
@@ -1083,6 +1089,22 @@ void HackStudioWidget::initialize_menubar(GUI::Menubar& menubar)
+    StringBuilder builder;
+    if (current_editor().has_selection()) {
+        auto line_count = current_editor().selected_line_count();
+        builder.appendff("Selected: {} {}", line_count, line_count != 1 ? "Lines" : "Line");
```
```suggestion
        builder.appendff("Selected: {} {}", line_count, line_count != 1 ? "lines" : "line");
```

linusg commented on Userland/Libraries/LibGUI/TextEditor.h:
```diff
@@ -135,6 +135,8 @@ class TextEditor
     bool write_to_file(const String& path);
     bool has_selection() const { return m_selection.is_valid(); }
     String selected_text() const;
+    int selected_word_count() const;
+    size_t selected_line_count() const;
```
This API feels kinda weird - why are these different types, and why signed?
